### PR TITLE
Added implementation to use user provided access token for existing connections and generate new access token for new connections using refresh token

### PR DESCRIPTION
**Pull request by RushT007**
# Description of change
- Added implementation to use user provided access token for existing connections and generate new access token for new connections using refresh token
- Added tap-tester to test existing connection scenario i.e. no refresh token provided
- Added the unittest to test access token creation
- Updated existing unittests

# Manual QA steps
 - [x] Verify that sync works when refresh token in not provided in config properties
 - [x] Verify the discovery fails when token is expired or invalid
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Review by KrisPersonal — APPROVED** [review]

File: tap_linkedin_ads/__init__.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env python3
 
+from os import access
```
Comment (cosimon) [inline tap_linkedin_ads/__init__.py:3]: Do we need this import?

File: tests/unittests/test_timeout.py [hunk 2]
```diff
@@ -228,17 +246,19 @@ def test_connection_error__check_access_token(self, mocked_request, mocked_sleep
...
         }
 
         # initialize 'LinkedinClient'
         try:
             with client.LinkedinClient(client_id=config['client_id'],
-        client_secret=config['client_secret'],
-        refresh_token=config['refresh_token'],
-        user_agent=config['user_agent'],
-        request_timeout=config.get('request_timeout')) as cl:
+            client_secret=config['client_secret'],
```
Comment (cosimon) [inline tests/unittests/test_timeout.py:257]: Please align these correctly with the `client_id`

**Review by cosimon — CHANGES_REQUESTED** [review]

**Comment by RushT007** [comment]
@cosimon I have fixed all the issues and also successfully run the alpha for internal stitch client.

File: tap_linkedin_ads/client.py [hunk 3]
```diff
@@ -159,10 +165,14 @@ def __exit__(self, exception_type, exception_value, traceback):
...
                           factor=2)
     def fetch_and_set_access_token(self):
-        """"""
-        # The refresh_token never expires and may be used many times to generate each access_token
-        # Since the refresh_token does not expire, it is not included in get access_token response
-        if self.__access_token is not None and self.__expires > datetime.utcnow():
+        """
+        This method generates new access token if the refresh token is provided.
+
+        Note: Linkedin-ads access token expires in 90 days, where as refresh token expires in 365 days.
```
Comment (cosimon) [inline tap_linkedin_ads/client.py:171]: Access tokens expire in 60 days, not 90. Please update this comment.

File: tests/unittests/test_timeout.py [hunk 4]
```diff
@@ -50,16 +56,19 @@ def test_timeout_string_value_passed_in_config(self):
...
-        refresh_token=config['refresh_token'],
-        user_agent=config['user_agent'],
-        request_timeout=config.get('request_timeout'))
+        cl = client.LinkedinClient(
+            client_id=config['client_id'],
+            client_secret=config['client_secret'],
+            refresh_token=config['refresh_token'],
+            access_token=config['access_token'],
+            user_agent=config['user_agent'],
+            request_timeout=config.get('request_timeout'))
```
Comment (cosimon) [inline tests/unittests/test_timeout.py:71]: Please align them like this:
```suggestion
        cl = client.LinkedinClient(client_id=config['client_id'],
                                   client_secret=config['client_secret'],
                                   refresh_token=config['refresh_token'],
                                   access_token=config['access_token'],
                                   user_agent=config['user_agent'],
                                   request_timeout=config.get('request_timeout'))
```

**Review by cosimon — CHANGES_REQUESTED** [review]

File: tests/unittests/test_timeout.py (hunk 4 above)
Comment (RushT007) [reply tests/unittests/test_timeout.py]: Made the changes, please re-review.

**Review by cosimon — APPROVED** [review]

**Pull request merged**